### PR TITLE
Convert gcloud operations in storage client - 1

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Remove the gcloud commands for creating buckets and cleaning up directories, and replace them with storage client functions. This is because running gcloud commands in parallel sometimes leads to failures.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - Automated, Ran tests with mountedDirectory flag.

### CODE REVIEW

**Review by ashmeenkaur — COMMENTED** [review]
Did we run the tests with mountedDirectory flag too?

File: tools/integration_tests/util/mounting/dynamic_mounting/dynamic_mounting.go [hunk 1]
```diff
@@ -98,33 +102,71 @@ func executeTestsForDynamicMounting(flags [][]string, m *testing.M) (successCode
...
+	if err := bucket.Delete(ctx); err != nil {
+		log.Printf("Bucket(%q).Delete: %v", bucketName, err)
+	}
 }
 
 func RunTests(flags [][]string, m *testing.M) (successCode int) {
 	log.Println("Running dynamic mounting tests...")
 
-	CreateTestBucketForDynamicMounting()
+	ctx := context.Background()
```
Comment (ashmeenkaur) [inline tools/integration_tests/util/mounting/dynamic_mounting/dynamic_mounting.go:153]: Let's reuse the common storage client created at test level

**Comment by Tulsishah** [comment]
> Did we run the tests with mountedDirectory flag too?

Ran all the things are working fine.

File: tools/integration_tests/operations/operations_test.go [hunk 2]
```diff
@@ -87,6 +92,11 @@ const ContentInFileInDirThreeInCreateThreeLevelDirTest = "Hello world!!"
 const Content = "line 1\nline 2\n"
 const onlyDirMounted = "OnlyDirMountOperations"
 
+var (
```
Comment (ashmeenkaur) [inline tools/integration_tests/operations/operations_test.go:95]: Nit: don't see a need for global variable.. same in other tests also

File: tools/integration_tests/util/mounting/dynamic_mounting/dynamic_mounting.go [hunk 3]
```diff
@@ -98,33 +101,61 @@ func executeTestsForDynamicMounting(flags [][]string, m *testing.M) (successCode
...
+		}
+	}
+
+	if err := bucket.Delete(ctx); err != nil {
+		log.Printf("Bucket(%q).Delete: %v", bucketName, err)
+	}
 }
 
-func RunTests(flags [][]string, m *testing.M) (successCode int) {
+func RunTests(flags [][]string, ctx context.Context, client *storage.Client, m *testing.M) (successCode int) {
```
Comment (ashmeenkaur) [inline tools/integration_tests/util/mounting/dynamic_mounting/dynamic_mounting.go:149]: Make context first parameter: https://google.github.io/styleguide/go/decisions#contexts

File: tools/integration_tests/util/mounting/dynamic_mounting/dynamic_mounting.go [hunk 4]
```diff
@@ -98,33 +101,61 @@ func executeTestsForDynamicMounting(flags [][]string, m *testing.M) (successCode
...
+		}
+
+		obj := bucket.Object(objAttrs.Name)
+		err = obj.Delete(ctx)
+		if err != nil {
+			log.Fatalf("Failed to delete object %s: %v", objAttrs.Name, err)
+		}
+	}
+
+	if err := bucket.Delete(ctx); err != nil {
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply tools/integration_tests/util/mounting/dynamic_mounting/dynamic_mounting.go:144]: No we can't, we have to recursively delete the bucket.

File: tools/integration_tests/read_cache/remount_test.go [hunk 5]
```diff
@@ -91,12 +91,12 @@ func (s *remountTest) TestCacheIsNotReusedOnDynamicRemount(t *testing.T) {
...
 	testFileName1 := setupFileInTestDir(s.ctx, s.storageClient, testDirName, fileSize, t)
-	testBucket2 := dynamic_mounting.CreateTestBucketForDynamicMounting()
-	defer dynamic_mounting.DeleteTestBucketForDynamicMounting(testBucket2)
+	testBucket2 := dynamic_mounting.CreateTestBucketForDynamicMounting(ctx, storageClient)
+	defer dynamic_mounting.DeleteTestBucketForDynamicMounting(ctx, storageClient, testBucket2)
 	setup.SetDynamicBucketMounted(testBucket2)
 	defer setup.SetDynamicBucketMounted("")
-	// Introducing a sleep of 7 seconds after bucket creation to address propagation delays.
-	time.Sleep(7 * time.Second)
+	// Introducing a sleep of 10 seconds after bucket creation to address propagation delays.
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply tools/integration_tests/read_cache/remount_test.go:98]: Yes for 7 seconds in was intermittently failing.

File: tools/integration_tests/interrupt/interrupt_test.go [hunk 6]
```diff
@@ -28,13 +33,26 @@ const (
 	testDirName = "InterruptTest"
 )
 
+var (
+	storageClient *storage.Client
```
Comment (ashmeenkaur) [inline tools/integration_tests/interrupt/interrupt_test.go:37]: I don't think we need this global variable

File: tools/integration_tests/log_rotation/log_rotation_test.go [hunk 7]
```diff
@@ -36,8 +41,12 @@ const (
 	logFileCount       = activeLogFileCount + backupLogFileCount
 )
 
-var logDirPath string
-var logFilePath string
+var (
+	logDirPath    string
+	logFilePath   string
+	storageClient *storage.Client
```
Comment (ashmeenkaur) [inline tools/integration_tests/log_rotation/log_rotation_test.go:47]: This too

File: tools/integration_tests/interrupt/interrupt_test.go (hunk 6 above)
Comment (Tulsishah) [reply tools/integration_tests/interrupt/interrupt_test.go]: Done

**Pull request merged**